### ISSUE TITLE
Multi-gpu memory problem.

### ISSUE THREAD

**Issue by happsky** [junyanz/pytorch-CycleGAN-and-pix2pix#545]
I use two gpus to train the model with batchsize 2, and one gpu costs about 10G memory, while another one only consumes around 3G memory. My question is that can I balance the memory used in these two gpus so that each one costs about 6G memory?

When I am trying to use 4 gpus to train the model with batchsize 4, I failed. Seems to be the first gpu memory overflow. Any suggestions to fix both issues?

**Comment by taesungp** [comment]
I think it's because GPU0 also needs to do the work of combining the gradients. You will probably need to modify the code inside pytorch itself to change it. This code repo does not plan to address this issue. The authors did not use multi-GPU training.
